src/Database/LSMTree/Internal/MergeSchedule.hs

```diff
@@ -683,15 +703,20 @@ addRunToLevels tr conf@TableConfig{..} resolve hfs hbio root uc r0 reg levels =
     expectCompletedMerge ln (SingleRun r) = do
       traceWith tr $ AtLevel ln $ TraceExpectCompletedMergeSingleRun (runNumber $ Run.runRunFsPaths r)
       pure r
-    expectCompletedMerge ln (MergingRun var) = do
-      withMVar var $ \case
-        CompletedMerge r -> do
+    expectCompletedMerge ln (MergingRun _ _ var) = do
+      modifyMVarMasked var $ \case
+        x@(CompletedMerge r) -> do
           traceWith tr $ AtLevel ln $ TraceExpectCompletedMerge (runNumber $ Run.runRunFsPaths r)
-          pure r
-        OngoingMerge _rs _m -> error "expectCompletedMerge: OngoingMerge not yet supported" -- TODO: implement.
+          pure (x, r)
+        OngoingMerge _rs _ _m -> do
+          -- RefCount n <- Merge.readRefCount m
+          -- let !n' = fromIntegralChecked n
+          -- V.forM_ rs $ \r -> Run.removeReferenceN r n'
+          -- r <- Merge.complete m
+          -- Merge.removeReferenceN m n'
+          -- pure (CompletedMerge r, r)
+          error "expectCompletedMerge: OngoingMerge not yet supported" -- TODO: implement.
```

Comment:
This should replace by a proper error, or it should instead become a warning (e.g., a trace message) and we step the merge to completion here. I'm not sure which is best. If we call `expectCompletedMerge` on a merging run and it is not completed, then it *is* actually signalling that we have a bug in our scheduling

Reply:
Moved into a TODO in the code.